docs/architecture/blazor-for-web-forms-developers/data.md

```diff
@@ -1,22 +1,132 @@
+
+### Database First
+
+For existing databases, you can generate the classes for EF Core by using the .NET command-line tools. To scaffold the classes, use a variation of the following command:
+
+```dotnetcli
+dotnet ef dbcontext scaffold "CONNECTION STRING" Microsoft.EntityFrameworkCore.SqlServer -c MyDbContext -t Product -t Customer
+```
+
+The preceding command connects to the database using the specified connection string and the `Microsoft.EntityFrameworkCore.SqlServer` provider. Once connected, a database context class named `MyDbContext` is created. Additionally, supporting classes are created for the `Product` and `Customer` tables that were specified with the `-t` options. There are many configuration options for this command to generate the class hierarchy appropriate for your database. For a complete reference, see [the command's documentation](/ef/core/miscellaneous/cli/dotnet#dotnet-ef-dbcontext-scaffold).
+
+More information about [EF Core](/ef/core/) can be found on the Microsoft Docs site.
+
+## Interact with web services
+
+When ASP.NET was first released, SOAP services were the preferred way for web servers and clients to exchange data. Much has changed since that time, and the preferred interactions with services have shifted to direct HTTP client interactions. With ASP.NET Core and Blazor, you can register the configuration of your `HttpClient` in the `Startup` class's `ConfigureServices` method. Use that configuration when you need to interact with the HTTP endpoint. Consider the following configuration code:
+
+```csharp
+services.AddHttpClient("github", c =>
+{
+    c.BaseAddress = new Uri("http://api.github.com/");
+    // Github API versioning
+    c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+    // Github requires a user-agent
+    c.DefaultRequestHeaders.Add("User-Agent", "BlazorWebForms-Sample");
+});
+```
+
+Whenever you need to access data from GitHub, create a client with a name of `github`. The client is configured with the base address, and the request headers are set appropriately. Inject the `IHttpClientFactory` into your Blazor components with the `@inject` directive or an `[Inject]` attribute on a property. Create your named client and interact with services using the following syntax:
+
+```razor
+
+@inject IHttpClientFactory factory
+
+...
+
+@code {
+    protected override async Task OnInitializedAsync()
+    {
+        var client = factory.CreateClient("github");
+        var response = await client.GetAsync("repos/dotnet/docs/issues");
+        response.EnsureStatusCode();
+        var content = async test.Content.ReadAsStringAsync();
```

Comment:
```suggestion
        var content = async response.Content.ReadAsStringAsync();
```

My brains' compiler said, error! `test` does not exist in this context. Also, should we show some sort of interaction with the `content` variable? It kind of feels neglected, I know that the paragraph afterwards talks about it - but when I see this as a C# developer, I think this should be a discard `_ = async response.Content.ReadAsStringAsync();` which is wrong here.